src/core/integrate.cpp

```diff
@@ -148,41 +152,109 @@ template <class Kernel> void run_kernel(BoxGeometry const &box_geo) {
 }
 } // namespace LeesEdwards
 
+static void update_default_propagation() {
+  switch (integ_switch) {
+  case INTEG_METHOD_STEEPEST_DESCENT:
+    default_propagation = PropagationMode::NONE;
+    break;
+  case INTEG_METHOD_NVT: {
+    // NOLINTNEXTLINE(bugprone-branch-clone)
+    if (thermo_switch & THERMO_LB & THERMO_LANGEVIN) {
+      default_propagation = PropagationMode::TRANS_LB_MOMENTUM_EXCHANGE;
```

Comment:
This conditional branch activates translation LB + rotation Langevin. Maybe I misinterpreted your request for implementing the mixed case "LB viscous coupling for translation + langevin friction for rotation"?

Reply:
Ah, OK. How do we prevent the following then:
system wide Langeivn an dLB thermostats with a non-zero gamma translaiton on the Langevin thermostat and
particle.propagaoi = systej default
This would then apply both translational Langevin noise and frictoin and couple the translational degree of feeom to LB, which would be wrong.

Reply:
good point, I have added an extra conditional branch to cover that exact use case